File: PREGUNTASYRESPUESTAS/Backend/Backend/Backend/Domain/Models/Cuestionario.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Domain.Models
{
    public class Cuestionario
    {
        public int Id { get; set; }
        [Required]
        [Column (TypeName ="varchar(100)")]
        public required string Nombre{ get; set; }

        [Required]
        [Column(TypeName = "varchar(150)")]
        public required string Descripcion { get; set; }

        public DateTime FechaCreacion { get; set; } = DateTime.Now;
        public int Activo { get; set; }
        public int UsuarioId { get; set; }
        public  Usuario Usuario { get; set; }
        public  List<Pregunta> listPreguntas { get; set; }
    }
}

File: PREGUNTASYRESPUESTAS/FrontEnd/src/app/models/cuestionario.ts
import { Pregunta } from "./pregunta";
import { Usuario } from "./usuario";

export class Cuestionario{
  id?: number;
  nombre: string;
  descripcion: string;
  fechaCreacion?: Date;
  listPreguntas: Pregunta[];
  usuario?: Usuario;

  constructor(nombre: string, description: string,fechaCreacion: Date,listPreguntas: Pregunta[], usuario: Usuario ){
    this.nombre = nombre;
    this.descripcion = description;
    this.fechaCreacion = fechaCreacion;
    this.listPreguntas =listPreguntas;
    this.usuario = usuario;
  }
}
